activesupport/lib/active_support/callbacks.rb

```diff
@@ -826,6 +754,30 @@ def get_callbacks(name)
       def set_callbacks(name, callbacks)
         send "_#{name}_callbacks=", callbacks
       end
+
+      def default_false_terminator
+        Proc.new do |target, result_lambda|
+          terminate = true
+          catch(:abort) do
+            result = result_lambda.call if result_lambda.is_a?(Proc)
+            if CallbackChain.halt_and_display_warning_on_return_false && result == false
```

Comment:
Agreed, except for:

> It also sounds like using Active Support callbacks directly is an edge case, so in our deprecation message it's fine to treat AR and AM callbacks as all the callbacks IMO.

It's not the edge case. Rails itself uses AS callbacks directly, as well as third-party gems.